docs/DeveloperGuide.md

```diff
@@ -1,15 +1,93 @@
 # Developer Guide
+Welcome to the NUSPlanner Developer Guide!
+We hope this documents serves useful to understand the behind-the-scenes working of our product. 
+
+## Table of Contents
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}'
+
+## Getting started
+
+Refer to the user guide [here](https://github.com/AY2223S2-CS2113-F13-3/tp/blob/master/docs/UserGuide.md).
+
 
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Parser Component
+The component will return the correct command based on what the user inputs into the application.
+
+#### How is the feature implemented?
+
+The Parser component parses the command of the user input and breaks the user input into different parts based on the flags.
+This component also ensures to validate that user input is correct.
+
```

Comment:
![image](https://user-images.githubusercontent.com/88603534/228765320-97982f7b-e6f0-464f-8d54-5473197b0261.png)
This sequence diagram is missing activation bars, it would be better if you include them